resolwe_runtime_utils.py

```diff
@@ -48,51 +137,168 @@ def _get_json(value):
         return json.loads('"{}"'.format(value))
 
 
-def save(key, value):
-    """Convert the given parameters to a JSON object.
+def command(name, data):
+    """Create a command from command name and payload."""
+    return {"type": "COMMAND", "type_data": name, "data": data}
+
 
-    JSON object is of the form:
-    { key: value },
-    where value can represent any JSON object.
+def send_data(s, data, size_bytes=5):
+    """Send data over socket.
 
+    :param s: socket to send over.
+    :param data: dict that must be serialzable to JSON.
+    :param size_bytes: how first many bytes in message are dedicated to the
+        message size (pre-padded with zeros).
+    :raises: exception on failure.
     """
-    return json.dumps({key: _get_json(value)})
 
+    def _check_response(response):
+        if response is None:
+            raise RuntimeError(
+                "No response received when sending message: {}.".format(data)
+            )
 
-def save_list(key, *values):
-    """Convert the given list of parameters to a JSON object.
+        if "type_data" not in response:
+            raise ValueError(
+                "Response {} does not contain key 'status'.".format(response)
+            )
 
-    JSON object is of the form:
-    { key: [values[0], values[1], ... ] },
-    where values represent the given list of parameters.
+        if response["type_data"] != "OK":
+            return False
+        return True
+
+    message = json.dumps(data).encode("utf-8")
```

Comment:
Why do you have `send_data` and `send_json_message` functions if you dump the data to JSON in the first one?

Reply:
I have merged them and renamed them to `send_message`.